DatadogRUM/Sources/Integrations/WebViewEventReceiver.swift

```diff
@@ -111,6 +115,38 @@ internal final class WebViewEventReceiver: FeatureMessageReceiver {
         }
     }
 
+    private func receive(telemetry event: JSON, from core: DatadogCoreProtocol) {
+        // RUM-2866: Update with dedicated telemetry track
+        featureScope.eventWriteContext { [weak core] context, writer in
+            guard let rumBaggage = context.baggages[RUMFeature.name] else {
+                return // Drop event if RUM is not enabled or RUM session is not sampled
+            }
+
+            do {
+                let rum: RUMCoreContext = try rumBaggage.decode()
+                var event = event
+
+                if let date = event["date"] as? Int {
+                    event["date"] = Int64(date) + context.serverTimeOffset.toInt64Milliseconds
+                }
+
+                if var application = event["application"] as? JSON {
+                    application["id"] = rum.applicationID
+                    event["application"] = application
+                }
+
+                if var session = event["session"] as? JSON {
+                    session["id"] = rum.sessionID
+                    event["session"] = session
+                }
+
+                writer.write(value: AnyEncodable(event))
+            } catch {
+                core?.telemetry.error("Failed to decode `RUMCoreContext`", error: error)
```

Comment:
change-request/ The telemetry endpoint is now available on `featureScope` and that can be even captured strongly to the closure (it manages weak reference to `core`):
```swift
featureScope.eventWriteContext { [featureScope] context, writer in
   // ...
   featureScope.telemetry.error("...")
}
```